client/lib/site/utils.js

```diff
@@ -176,3 +178,16 @@ export function hasSiteFeature( site, feature ) {
 		return planHasFeature( site.plan.product_slug, feature );
 	}
 }
+
+/**
+ * Checks if the site has a free plan and is Atomic.
+ *
+ * @param {object} state  Global state tree
+ * @param {number} siteId Site ID
+ * @returns {boolean} True if site is free and Atomic
+ */
+export function isFreeAtomicSite( state, siteId ) {
+	const isAtomic = isAtomicSite( state, siteId );
```

Comment:
Does it make sense to bail early when it isn't an Atomic site? Something like...
```suggestion
	const isAtomic = isAtomicSite( state, siteId );
	if ( ! isAtomic ) {
	  return false;
  }
```